src/main/java/net/fabricmc/loom/configuration/providers/mappings/LayeredMappingsDependency.java

```diff
@@ -102,7 +103,8 @@ private void writeMapping(LayeredMappingsProcessor processor, List<MappingLayer>
 
 			MappingDstNsReorder nsReorder = new MappingDstNsReorder(tiny2Writer, Collections.singletonList(MappingsNamespace.NAMED.toString()));
 			MappingSourceNsSwitch nsSwitch = new MappingSourceNsSwitch(nsReorder, MappingsNamespace.INTERMEDIARY.toString(), true);
-			mappings.accept(nsSwitch);
+			AddConstructorMappingVisitor addConstructor = new AddConstructorMappingVisitor(nsSwitch);
```

Comment:
Would this be better in `MojangMappingLayer`?

Reply:
I don't exactly see how this could be moved to MojangMappingLayer, and the main cause of the issue is that intermediary mappings don't include <init>, not mojang mappings